pgcli/pgcompleter.py

```diff
@@ -45,12 +46,31 @@ def __init__(self, smart_completion=True):
         super(self.__class__, self).__init__()
         self.smart_completion = smart_completion
 
+        self.name_pattern = compile("^[_a-z][_a-z0-9\$]*$")
+
+    def extend_escape_name(self, name):
+        if not self.name_pattern.match(name) or name in self.keywords or name in self.functions:
```

Comment:
The lookups inside keywords and functions has to be case-insensitive. Because someone could name their columns `select` all lower case, which should be escaped, but it won't be because the keywords is a collection of uppercased words.

So change this check to: 
if not self.name_pattern.match(name) or name.lower() in self.keywords or name.lower() in self.functions: